app/src/main/java/com/example/bigowlapp/repository/Repository.java

```diff
@@ -131,6 +132,31 @@ public boolean removeDocument(String docUId) {
         return tData;
     }
 
+    public MutableLiveData<List<T>> getDocumentsByUId(List<String> uIdList, Class<? extends T> tClass) {
```

Comment:
I would rename this method to getDocumentsByListOfUId for more clarity.

The current name implies that you could get multiple documents using a single UId. Just a suggestion if you think it's not necessary to change, let me know.

Reply:
Makes sense, will do